### ISSUE TITLE
Training LLAVA-1.5-7B on LLAVA-1.5 Dataset: Loss Becomes 0

### ISSUE THREAD

**Issue by showstarpro** [SaraGhazanfari/EMMA#2]
Dear authors,
Thank you for sharing your excellent work! I am trying to train LLAVA-1.5-7B using the LLAVA-1.5 dataset following the settings described in your paper. However, I encountered an issue where the loss becomes 0 during training.

Could you please clarify the exact training settings you used for this experiment? Specifically:

Hyperparameters (batch size, learning rate, etc.)
Any modifications to the codebase or additional pre-processing steps required for LLAVA-1.5 dataset compatibility
Steps I Followed
Dataset: Used the LLAVA-1.5 dataset.
Model: LLAVA-1.5-7B.
Training setup:
#!/bin/bash

# IMPORTANT: this is the training script for the original LLaVA, NOT FOR LLaVA V1.5!

deepspeed llava/train/train_mem.py \
    --deepspeed ./scripts/zero2.json \
    --model_name_or_path /lpai/volumes/so-volume-ga/models/vicuna-7b-v1.5 \
    --version plain \
    --data_path /lpai/dataset/llava-pre/0-1-0/LLaVA-Pretrain/blip_laion_cc_sbu_558k.json \
    --image_folder /lpai/dataset/llava-pre/0-1-0/LLaVA-Pretrain/images \
    --vision_tower /lpai/volumes/so-volume-ga/models/clip-vit-large-patch14-336 \
    --tune_mm_mlp_adapter True \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --bf16 True \
    --output_dir /lpai/volumes/so-volume-ga/lhp/vicuna-7b-v1.5-pretrain/emma_base \
    --num_train_epochs 1 \
    --per_device_train_batch_size 16 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 24000 \
    --save_total_limit 1 \
    --learning_rate 2e-3 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 2048 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --lazy_preprocess True \
    --report_to wandb \
    --mm_text_select_layer -2 \
    --num_learnable_tokens 0 \
    --encoder_version v1 \
    --mm_text_select_feature patch  

################## VICUNA ##################
# PROMPT_VERSION=v1
# MODEL_VERSION="vicuna-v1-3-7b"
################## VICUNA ##################

deepspeed llava/train/train_mem.py \
    --deepspeed ./scripts/zero2.json \
    --model_name_or_path /lpai/volumes/so-volume-ga/models/vicuna-7b-v1.5 \
    --version v1 \
    --data_path /lpai/dataset/llava-ft/0-1-4/llava_ft/llava_v1_5_mix665k.json  \
    --image_folder /lpai/dataset/llava-ft/0-1-4/llava_ft/data \
    --vision_tower /lpai/volumes/so-volume-ga/models/clip-vit-large-patch14-336 \
    --pretrain_mm_mlp_adapter /lpai/volumes/so-volume-ga/lhp/vicuna-7b-v1.5-pretrain/emma_base/mm_projector.bin \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --bf16 True \
    --output_dir /lpai/volumes/so-volume-ga/lhp/llava-v1.5/vicuna-7b-v1.5-pretrain/llava-v1.5-7b-emma_base \
    --num_train_epochs 1 \
    --per_device_train_batch_size 16 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 50000 \
    --save_total_limit 1 \
    --learning_rate 2e-5 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 2048 \
    --gradient_checkpointing True \
    --dataloader_num_workers 4 \
    --lazy_preprocess True \
    --report_to wandb \
    --mm_text_select_layer -2 \
    --num_learnable_tokens 0 \
    --encoder_version v1 \
    --mm_text_select_feature patch  
![image](https://github.com/user-attachments/assets/3c472bde-38c6-4a16-9034-d44a76d6f677)

**Comment by EchoDreamer** [comment]
I have encountered the same problem. Have you solved it? I also found some problems with this code. Maybe we can discuss it with each other.

**Comment by showstarpro** [comment]
Sorry, I did not sovle this problem.
